src/ConstantSum/ConstantSumMath.sol

```diff
@@ -24,77 +25,28 @@ function computeInitialPoolData(
 ) pure returns (bytes memory) {
     // The pool can be initialized with any non-negative amount of rx, and ry.
     // so we have to allow a user to pass an amount of both even if one is zero.
-    uint256 L = rx + ry.divWadUp(params.price);
     uint256[] memory reserves = new uint256[](2);
     reserves[0] = rx;
     reserves[1] = ry;
-    return abi.encode(reserves, L, params);
+    return abi.encode(reserves, params);
 }
 
-function computeDeallocateGivenDeltaX(
+function computeDeltaLiquidity(
     uint256 deltaX,
-    uint256 rX,
-    uint256 rY,
-    uint256 totalLiquidity
-) pure returns (uint256 deltaY, uint256 deltaL) {
-    uint256 a = deltaX.divWadDown(rX);
-    if (rY > 0) {
-        deltaY = a.mulWadDown(rY);
-    }
-    deltaL = a.mulWadDown(totalLiquidity);
-}
-
-function computeDeallocateGivenDeltaY(
-    uint256 deltaY,
-    uint256 rX,
-    uint256 rY,
-    uint256 totalLiquidity
-) pure returns (uint256 deltaX, uint256 deltaL) {
-    uint256 a = deltaY.divWadDown(rY);
-    if (rX > 0) {
-        deltaX = a.mulWadDown(rX);
-    }
-    deltaL = a.mulWadDown(totalLiquidity);
-}
-
-function computeAllocateGivenDeltaX(
-    uint256 deltaX,
-    uint256 rX,
-    uint256 rY,
-    uint256 totalLiquidity
-) pure returns (uint256 deltaY, uint256 deltaL) {
-    uint256 a = deltaX.divWadUp(rX);
-    if (rY > 0) {
-        deltaY = a.mulWadUp(rY);
-    }
-    deltaL = a.mulWadUp(totalLiquidity);
-}
-
-function computeAllocateGivenDeltaY(
     uint256 deltaY,
-    uint256 rX,
-    uint256 rY,
-    uint256 totalLiquidity
-) pure returns (uint256 deltaX, uint256 deltaL) {
-    uint256 a = deltaY.divWadUp(rY);
-    if (rX > 0) {
-        deltaX = a.mulWadUp(rX);
-    }
-    deltaL = a.mulWadUp(totalLiquidity);
-}
-
-function computeDeltaGivenDeltaLRoundUp(
-    uint256 reserve,
-    uint256 deltaLiquidity,
-    uint256 totalLiquidity
+    uint256 price
 ) pure returns (uint256) {
-    return reserve.mulWadUp(deltaLiquidity.divWadUp(totalLiquidity));
+    return price.mulWadUp(deltaX) + deltaY;
 }
 
-function computeDeltaGivenDeltaLRoundDown(
-    uint256 reserve,
-    uint256 deltaLiquidity,
-    uint256 totalLiquidity
+function computeSwapDeltaLiquidity(
+    uint256 delta,
+    ConstantSumParams memory params,
+    bool isSwapXForY
 ) pure returns (uint256) {
-    return reserve.mulWadDown(deltaLiquidity.divWadDown(totalLiquidity));
+    if (isSwapXForY) {
+        return (params.swapFee).mulWadUp(delta);
+    } else {
+        return (params.swapFee).mulDivUp(delta, params.price);
+    }
```

Comment:
This logic seems inconsistent with the other functions in this file (e.g. `computeDeltaLiquidity` immediately above). Liquidity is in units of `Y` and `X` quantities need to be multiplied by price. If `isSwapXForY == true`, I expect `delta` to be an `X` amount and thus the liquidity change should be something like `(params.swapFee).mulWadUp(delta.mulWadUp(params.price))`, and in the `false` case, it should be `(params.swapFee).mulWadUp(delta)` as then `delta` is a `Y` quantity.

Reply:
Can confirm this, when calling `swap()` with the corrent input and output amounts, you either blow-up the invariant or it becomes negative:

```solidity
// SPDX-License-Identifier: UNLICENSED
pragma solidity 0.8.18;

import "forge-std/Test.sol";
import "src/FixedPointMathLib.sol";

contract SwapTest is Test {
    using FixedPointMathLib for uint256;
    using FixedPointMathLib for int256;

    function testBounds() public {
        // Pool state
        uint256 rX = 10e18;
        uint256 rY = 20e18;
        uint256 price = 2e18;
        uint256 swapFee = 0.05e18;

        uint256 tL = price.mulWadUp(rX) + rY;
        _invariant(rX, rY, tL, price);

        // Swap 1 tokenX for 2 tokenY
        uint256 amountIn = 1e18;
        uint256 amountOut = uint256(2e18).mulWadDown(1e18 - swapFee);
        uint256 dL = computeSwapDeltaLiquidity(amountIn, price, swapFee, false);

        _invariant(
            rX + amountIn,
            rY - amountOut,
            tL + dL,
            price
        );
    }

    function _invariant(
        uint256 rX,
        uint256 rY,
        uint256 tL,
        uint256 price
    ) internal returns (int256) {
        int256 invariant = int256(price.mulWadUp(rX.divWadUp(tL)) + rY.divWadUp(tL)) - int256(1e18);
        console2.log("invariant:", invariant);
        assertTrue(invariant >= 0);
    }

    function computeSwapDeltaLiquidity(
        uint256 delta,
        uint256 price,
        uint256 swapFee,
        bool isSwapXForY
    ) internal returns (uint256) {
        if (isSwapXForY) {
            return (swapFee).mulWadUp(delta);
        } else {
            return (swapFee).mulDivUp(delta, price);
        }
    }
}
```

Reply:
Fixed here https://github.com/primitivefinance/DFMM/pull/119/commits/6b5a3f524c139053a812f3e08ebf0623f346598e.